### PR TITLE
Omissions in types supported by both Desktop and Core - System.Data.SqlClient.SqlBulkCopyColumnMappingCollection

### PR DESCRIPTION
**Pull request by ViktorHofer**
Adjustments for System.Data.SqlClient.SqlBulkCopyColumnMappingCollection to correctly inherit from System.Collections.CollectionBase. For more info see here: https://github.com/dotnet/corefx/issues/15255#issuecomment-289550875

### CODE REVIEW

**Comment by danmosemsft** [comment]
@saurabh500 PTAL

**Comment by saurabh500** [comment]
@corivera can you take a look?

**Comment by saurabh500** [comment]
@ViktorHofer Why is the contract being changed?

**Comment by ViktorHofer** [comment]
@saurabh500 There were a few issues with my commits as I included one of another PR. Should be fine now.

**Comment by safern** [comment]
Working on fixing your CI failures.

**Comment by safern** [comment]
test Vertical netfx Build please
test Windows_NT Debug AllConfigurations please

(#17612)

**Comment by safern** [comment]
test Windows_NT Debug AllConfigurations Build please

**Comment by danmosemsft** [comment]
Please don't merge without @corivera's signoff

**Comment by saurabh500** [comment]
@ViktorHofer Is this a port from Desktop?

**Comment by ViktorHofer** [comment]
@saurabh500 Small parts of it, yes

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SqlBulkCopyColumnMappingCollection.cs [hunk 1]
```diff
@@ -104,50 +72,48 @@ private void AssertWriteAccess()
             }
         }
 
-        public void Clear()
+        new public void Clear()
```
Comment (stephentoub) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SqlBulkCopyColumnMappingCollection.cs]: Nit: please reorder `new public` to `public new`, here and elsewhere.

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SqlBulkCopyColumnMappingCollection.cs [hunk 2]
```diff
@@ -104,50 +72,48 @@ private void AssertWriteAccess()
...
             AssertWriteAccess();
-            _list.RemoveAt(index);
+            base.RemoveAt(index);
         }
 
         internal void ValidateCollection()
         {
             MappingSchema mappingSchema;
-            foreach (SqlBulkCopyColumnMapping a in _list)
+            foreach (SqlBulkCopyColumnMapping a in this)
```
Comment (stephentoub) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SqlBulkCopyColumnMappingCollection.cs]: This is going to allocate an enumerator where it didn't allocate one before.  But I'm assuming that level of perf optimization here isn't important, given the scenarios?

**Comment by ViktorHofer** [comment]
@stephentoub: Thanks for the review!

**Pull request merged**